_posts/2020-03-09-o-openid-connect-slow-kilka.md

```diff
@@ -0,0 +1,106 @@
+---
+layout:    post
+title:     "O OpenID Connect słów kilka"
+published: true
+date:      2020-03-09 08:00:00 +0100
+author:    mhoja
+tags:
+    - openid connect
+    - oauth 2
+    - keycloak
+    - uwierzytelnianie użytkownika
+---
+
+W nawiązaniu do mojego poprzedniego wpisu pt.:  
+[**"Keycloak - uwierzytelnianie i autoryzacja użytkownika w aplikacji Angular/Spring Boot"** (link](https://blog.consdata.tech/2020/02/01/keycloak-uwierzytelnianie-autoryzacja-springboot-angular.html)),  
+chciałbym krótko opisać standard [OpenID Connect](https://openid.net/connect/), który został wykorzystany podczas logowania do aplikacji przy użyciu serwera uwierzytelniania [Keycloak](https://www.keycloak.org/).
+
+## Wstęp
+
+Najpopularniejszymi standardami wykorzystywanymi do uwierzytelniania/autoryzacji są `OAuth 2.0`, `OpenID Connect` oraz `SAML`.  
+O OAuth 2.0 zostało już napisanych wiele artykułów, których nie ma sensu powielać. Jednak aby przedstawić OpenID Connect, musiałbym opisać [OAuth 2.0](https://oauth.net/2/) oraz [JWT](https://jwt.io/).
+
+W celu zapoznania się ze standardem OAuth 2.0 odeślę do artykułu pt.:  
+[**"OAuth 2.0 – jak działa / jak testować / problemy bezpieczeństwa"** (link](https://sekurak.pl/oauth-2-0-jak-dziala-jak-testowac-problemy-bezpieczenstwa/))  
+autorstwa Marcina Pioska z portalu [Sekurak](https://sekurak.pl/), w którym została opisana terminologia, sposoby pozyskiwania tokenu oraz zasada działania standardu.
+
+O JWT natomiast możemy przeczytać w dokumencie [RFC7519](https://tools.ietf.org/html/rfc7519).
+
+Dlaczego więc do integracji naszej aplikacji z serwerem uwierzytelniania Keycloak użyliśmy OpenID Connect?
+
+## Uwierzytelnianie a autoryzacja
+
+Jeśli mówimy o zabezpieczeniu zasobów i dostępem do nich, mówimy o takich pojęciach jak **uwierzytelnianie** (ang. authentication) oraz **autoryzacja** (ang. authorization).
+
+- `uwierzytelnianie` to proces polegający na potwierdzeniu tożsamości, czyli w skrócie - *kim jestem?*;
+- `autoryzacja` to proces nadawania uprawnień (dostępu do zasobu), czyli w skrócie - *co mogę zrobić?*.
+
+OAuth 2.0 według oficjalnej dokumentacji nie powinien służyć do uwierzytelniania, a jedynie do autoryzacji ([źródło](https://oauth.net/articles/authentication/)):
+> OAuth 2.0 is not an authentication protocol.
+
+Jeśli potrzebujemy mechanizmu pozwalającego na poprawne zaimplementowanie uwierzytelniania, z pomocą przychodzi OpenID Connect.
+
+## OpenID Connect
+
+OpenID Connect jest prostą warstwą tożsamości opartą na OAuth 2.0.  
+Umożliwia klientom weryfikację tożsamości użytkownika końcowego na podstawie uwierzytelnienia przeprowadzonego przez serwer autoryzacji, a także uzyskanie podstawowych informacji o profilu użytkownika. Rozszerza OAuth 2.0, umożliwiając **uwierzytelnianie stowarzyszone** (ang. federated authentication):
```

Comment:
Przeredagowałbym to zdanie, żeby uniknąć powtórzenia "użytkownika". Można trochę przemieszać szyk i powinno być ok :)